### PR TITLE
[8.0] add stock_no_extra_move by gurneyalex

### PR DESCRIPTION
**Pull request by scigghia**
as discussed here https://odoo-community.org/groups/logistics-21/logistics-46347

@gurneyalex

### CODE REVIEW

**Comment by rousseldenis** [comment]
@scigghia @gurneyalex I think there is a fix (for further versions) for that behaviour.
I'll look for it. Anyway, this can be a good module.

**Comment by rousseldenis** [comment]
Found : https://github.com/odoo/odoo/pull/17589

File: stock_no_extra_move/README.rst [hunk 1]
```diff
@@ -0,0 +1,84 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+==============
+{module_title}
```
Comment (rousseldenis) [inline stock_no_extra_move/README.rst]: Please complete this or follow OCA readme generator conventions https://github.com/OCA/maintainer-tools/pull/339

File: stock_no_extra_move/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    "summary": "Prevent creation of extra moves in picking processing",
+    "version": "8.0.1.0.0",
+    "category": "Logistics",
+    "website": "https://camptocamp.com/",
+    "author": "Camptocamp",
+    "license": "AGPL-3",
+    "application": False,
+    "installable": True,
+    "depends": [
+        'specific_stock',
```
Comment (rousseldenis) [inline stock_no_extra_move/__openerp__.py]: What's this module ?

File: stock_no_extra_move/models/stock_picking.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    @api.model
+    def _create_extra_moves(self, picking):
+        User = self.pool['res.users']
+        can_create = (
+            User.has_group(
+                self.env.cr, self.env.user.id,
+                'specific_stock.group_utb_no_quantity_check'
```
Comment (rousseldenis) [inline stock_no_extra_move/models/stock_picking.py]: I think this module depends on non existing OCA module.

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review. Needs change

I would have also put a flag on picking types to allow or not extra moves as sometimes it is needed. e.g.: If your supplier sent you more products than ordered, or...

**Comment by scigghia** [comment]
@rousseldenis  incoming moves are not subjected to this check :)

**Comment by scigghia** [comment]
@rousseldenis could you take a look?

File: stock_no_extra_move/README.rst (hunk 1 above)
Comment (scigghia) [reply stock_no_extra_move/README.rst]: Done

File: stock_no_extra_move/__openerp__.py (hunk 2 above)
Comment (scigghia) [reply stock_no_extra_move/__openerp__.py]: removed

File: stock_no_extra_move/models/stock_picking.py (hunk 3 above)
Comment (scigghia) [reply stock_no_extra_move/models/stock_picking.py]: changed

**Pull request closed without merging**